fsspec/spec.py

```diff
@@ -429,6 +432,8 @@ def walk(self, path, maxdepth=None, topdown=True, **kwargs):
         if maxdepth is not None:
             maxdepth -= 1
             if maxdepth < 1:
+                if not topdown:
+                    yield path, dirs, files
```

Comment:
I have should have put this in PR #1128 but it only came to light when writing all the tests for this PR.

Reply:
It's a _good_ thing that these are coming out, thank you.